WordPress/Classes/ViewRelated/Menus/MenusViewController.m

```diff
@@ -176,6 +173,15 @@ - (void)viewWillDisappear:(BOOL)animated
     [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
+- (void)viewDidDisappear:(BOOL)animated
+{
+    [super viewDidDisappear:animated];
+
+    if (self.navigationController == nil && self.fetchAllPagesCancellationHandler != nil) {
```

Comment:
Why does the code check for the `navigationController` value?

Reply:
To make sure we only cancel the fetching when user taps the Back button. `viewDidDisappear` is also called when a new view controller is pushed into the stack. In which case, we'd like the pages fetching to continue.